rest/processing/process.py

```diff
@@ -104,21 +108,33 @@ def get_evalscript(self):
         evalscript = results[0]["evalscript"]
         evalscript.mosaicking = self.get_appropriate_mosaicking()
 
-        if self.get_input_bands() is None:
-            all_bands = collection["cube:dimensions"]["bands"]["values"]
+        if all(
+            bnds is None
+            for bnds in [datasource_with_bands["bands"] for datasource_with_bands in self.get_input_bands()]
+        ):
+            all_bands = []
+            for node_id, load_collection_node in load_collection_nodes.items():
+                collection = collections.get_collection(load_collection_node["arguments"]["id"])
+                all_bands.append(
+                    {"datasource": f"node_{node_id}", "bands": collection["cube:dimensions"]["bands"]["values"]}
+                )
```

Comment:
I think this doesn't work if only some of the `load_collection` processes don't have the bands set.

Also, I think it would be clearer like

```python
datasources_bands = [datasource_with_bands["bands"] for datasource_with_bands in self.get_input_bands()]
    if all(bnds is None for bnds in datasources_bands):
        ...
```

Reply:
Nice find, this did not work correctly. Now I check if any of the load collection nodes' bands are `None` and if this is true, then I iterate over them and if there are selected bands, I use those and if bands are `None` then I get all bands for that collection

Reply:
This seem to work correctly now. Thanks